tests/unit_tests/ensemble_evaluator/test_ensemble_legacy.py

```diff
@@ -30,16 +30,20 @@ def test_run_legacy_ensemble(tmpdir, make_ensemble_builder, monkeypatch):
         )
         evaluator = EnsembleEvaluator(ensemble, config, 0)
         evaluator.start_running()
-        with Monitor(config) as monitor:
-            for e in monitor.track():
-                if e["type"] in (
-                    identifiers.EVTYPE_EE_SNAPSHOT_UPDATE,
-                    identifiers.EVTYPE_EE_SNAPSHOT,
-                ) and e.data.get(identifiers.STATUS) in [
-                    state.ENSEMBLE_STATE_FAILED,
-                    state.ENSEMBLE_STATE_STOPPED,
-                ]:
-                    monitor.signal_done()
+
+        async def _run_monitor():
+            async with MonitorAsync(config) as monitor:
+                async for e in monitor.track():
+                    if e["type"] in (
+                        identifiers.EVTYPE_EE_SNAPSHOT_UPDATE,
+                        identifiers.EVTYPE_EE_SNAPSHOT,
+                    ) and e.data.get(identifiers.STATUS) in [
+                        state.ENSEMBLE_STATE_FAILED,
+                        state.ENSEMBLE_STATE_STOPPED,
+                    ]:
+                        await monitor.signal_done()
+
+        get_running_loop().run_until_complete(_run_monitor())
```

Comment:
Can we mark the test as async instead, and just await it?

Reply:
No, it won't work. Apparently the problem is when creating `EnsembleEvaluator`, which hijacks the running loop for itself.